src/charm.py

```diff
@@ -723,6 +760,35 @@ def _on_run_migration_action(self, event: ActionEvent) -> None:
             return
         event.log("Successfully migrated the database.")
 
+    def _promtail_error(self, event: PromtailDigestError):
+        logger.error(event.message)
+        self.unit.status = BlockedStatus(event.message)
+
+    def _on_update_status_handle_logging(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            logger.info(
+                "Cannot connect to Kratos container. Log file check delayed to next status update."
+            )
+            return
+
+        if not self.unit.status == ActiveStatus():
+            logger.info(
+                "Kratos unit currently inactive. Log file check delayed to next status update."
+            )
+            return
+
+        log_file_info_list = self._container.list_files(path=self._log_path)
+
+        if not log_file_info_list:
+            logger.info(
+                "Log file has not been created. Log file check delayed to next status update."
+            )
+            return
+
+        if int(log_file_info_list[0].size) > int(self.config["log_max_size"]) * 1000000:
+            logger.info("Log file greater than allowed size. Restarting container.")
+            self._handle_status_update_config(event)
```

Comment:
This function does not restart the container. It can be confusing because the same variable is used for the container and for the service, but this function will only restart the service. I don't think you can restart the container through pebble. 

It is not easy to handle file rotation (you would have to move the file -> restart kratos -> delete the old file after a while?, in this case we would have to provide the whole folder to loki?). Perhaps it is best to let the container die when the disk limit is reached, unless we can provide a solution that we know works well and in all cases.

Reply:
so for now I'll remove the logrotation, and raise it as an issue.